File: 04 motor/stepper_motor/stepper_wave.c
#include <stdio.h> 
#include <stdlib.h>
#include <signal.h>
#include <wiringPi.h> // Include WiringPi library!

const int PINS[4] = {4,17,27,22};
int DELAY = 5;

void my_ctrl_c_handler(int sig){ // can be called asynchronously
	int x;
	for( x  = 0; x <4; x++)
		digitalWrite(PINS[x], LOW);
	printf("Stepper Motor Test End\n"); 
	exit(0);
}

void wave_drive_stepper(seq)
{
	int x;
	for(x = 0; x < 4; x++){
		if(x == seq){
		    digitalWrite(PINS[x], HIGH);
		}
		else{
		    digitalWrite(PINS[x], LOW);
		}
	}
	delay(DELAY);
}

int main(void)
{
    // Setup stuff:
	int x;
	int count = 0;
	signal(SIGINT, my_ctrl_c_handler);	//Ctrl + C Handler
    wiringPiSetupGpio(); // Initialize wiringPi -- using Broadcom pin numbers

	for(x = 0; x < 4; x++){
		pinMode(PINS[x], OUTPUT); 
	}

	while (1){
		wave_drive_stepper(count % 4);
		count++;
	}
    return 0;
}

File: 04 motor/stepper_motor/stepper_wave.py
#!/usr/bin/env python
#-*- coding: utf-8 -*-
#If this code works, it was written by Seunghyun Lee(www.bluebaynetworks.co.kr).
#If not, I don't know who wrote it

import RPi.GPIO as gpio
import time

PINS = [4,17,27,22]
SEQA = [(4,),(17,),(27,),(22,)]
DELAY = 0.05

# 웨이브 드라이브 모드에서 코일에 전류 공급
def wave_drive_stepper(seq):
	for pin in PINS:
		if pin in SEQA[seq]:
			gpio.output(pin, gpio.HIGH)
		else:
			gpio.output(pin, gpio.LOW)
	time.sleep(DELAY)

gpio.setmode(gpio.BCM)

# 4, 17, 27, 22 번 핀을 출력 모드로 설정
for pin in PINS:
	gpio.setup(pin, gpio.OUT)

index = 0
try:
    while True:
		wave_drive_stepper(index % 4)
		index += 1
except KeyboardInterrupt:
    gpio.cleanup()
	print "Stepper Motor Test End" 
